### PR TITLE
[CN-7840] Fix elixir unit tests

### PR DESCRIPTION
**Pull request by engj**
CN-7840

Unit tests were modified to work with more recent versions of sqlalchemy for UTs to pass.

Some UTs related to encryption and the allowcoloverride option were removed as these features are either broken or haven't been implemented.

Used https://github.com/Didacti/elixir as a guide.

Run `tox` to execute the unit tests.

### CODE REVIEW

File: tests/test_encryption.py [hunk 1]
```diff
@@ -1,100 +0,0 @@
...
-
-        # check that it is not further encrypted
-        assert p.password == 'r\\x9d\\xa8\\xb4\\x8d|\\xffp\\xf5\\x0e'
-
-        session.commit()
-        session.expunge_all()
-
-        p = Person.get_by(name='Jonathan LaCour')
-        assert p.password == 's3cr3tw0RD'
-
```
Comment (engj) [inline tests/test_encryption.py:100]: These tests are broken, and we're not using elixir's encryption, so this file was removed.

File: tests/test_options.py [hunk 2]
```diff
@@ -65,20 +67,21 @@ class MyEntity(Entity):
...
-        # Note that this test is bogus as you cannot just change a column this
-        # way since the mapper is already constructed at this point and will
-        # use the old column!!! This test is only meant as a way to check no
-        # exception is raised.
-        #TODO: provide a proper test (using autoloaded tables)
-        MyEntity._descriptor.add_column(Column('name', String(30),
-                                               default='test'))
+    # NOTE(je): allowcoloverride option does not appear to be implemented
+    # def test_allowcoloverride_true(self):
+    #
```
Comment (engj) [inline tests/test_options.py]: `allowcoloverride` doesn't appear to be supported (this UT raises NotImplementedError) in the library and it isn't used in hermosa, so this UT was commented out.

File: setup.py [hunk 3]
```diff
@@ -33,7 +33,7 @@
       url="http://elixir.ematia.de",
       license = "MIT License",
       install_requires = [
-          "SQLAlchemy >= 0.5.0"
+          "SQLAlchemy >= 1.0.17"
```
Comment (engj) [inline setup.py:36]: Bumped to make UTs pass.

File: tests/test_options.py [hunk 4]
```diff
@@ -65,21 +67,6 @@ class MyEntity(Entity):
...
-        setup_all()
-
-        # Note that this test is bogus as you cannot just change a column this
-        # way since the mapper is already constructed at this point and will
-        # use the old column!!! This test is only meant as a way to check no
-        # exception is raised.
-        #TODO: provide a proper test (using autoloaded tables)
-        MyEntity._descriptor.add_column(Column('name', String(30),
-                                               default='test'))
-
```
Comment (engj) [inline tests/test_options.py:82]: This test was removed as `allowcoloverride` doesn't appear to be implemented.

File: tests/test_options.py [hunk 5]
```diff
@@ -4,7 +4,8 @@
 
 from sqlalchemy import UniqueConstraint, create_engine, Column
 from sqlalchemy.orm import scoped_session, sessionmaker
-from sqlalchemy.exceptions import SQLError, ConcurrentModificationError
```
Comment (halam) [inline tests/test_options.py:7]: I assume `SQLError` is deprecated in the later version of SQLAlchemy in favor of StatementError?

Comment (engj) [reply tests/test_options.py:7]: Correct.

**Review by halam — APPROVED** [review]
LGTM

**Pull request merged**